app/controllers/incomeSources/manage/ManageObligationsController.scala

```diff
@@ -0,0 +1,222 @@
+        Logger("application").error(
+          s"[BusinessAddedObligationsController][handleRequest] - unable to find incomeSource by id: $incomeSourceId ")
+        if (isAgent) Future(itvcErrorHandlerAgent.showInternalServerError())
+        else Future(itvcErrorHandler.showInternalServerError())
+      }
+      else{
+        val addedBusinessName: String = if (mode == "SE") {
+          val businessDetailsParams = for {
+            addedBusiness <- user.incomeSources.businesses.find(x => x.incomeSourceId.contains(incomeSourceId))
+            businessName <- addedBusiness.tradingName
+          } yield (addedBusiness, businessName)
+          businessDetailsParams match {
+            case Some((_, name)) => name
+            case None => "Not Found"
+          }
+        }
+        else {
+          if (mode == "UK") "UK property"
+          else "Foreign property"
+        }
+
+        val (year1,year2) = taxYear.splitAt(4)
+        val currenttaxYearStart = dateService.getCurrentTaxYearEnd() - 1
+        if ((year1.toInt == currenttaxYearStart && year2.drop(1).toInt == currenttaxYearStart+1)
+          || (year1.toInt == currenttaxYearStart+1 && year2.drop(1).toInt == currenttaxYearStart+2)){
+          if (changeTo == "annual" || changeTo == "quarterly") {
+            getIncomeSourceId(mode, incomeSourceId) match {
+              case Left(error) => Logger("application").error(
+                s"[BusinessAddedObligationsController][handleRequest] - ${error.getMessage}")
+                if (isAgent) Future(itvcErrorHandlerAgent.showInternalServerError())
+                else Future(itvcErrorHandler.showInternalServerError())
+              case Right(value) =>
+                nextUpdatesService.getObligationsViewModel(value, showPreviousTaxYears = false) map { viewModel =>
+                  if (isAgent) Ok(obligationsView(viewModel, mode, addedBusinessName, taxYear, changeTo, isAgent, backUrl, postUrl))
+                  else Ok(obligationsView(viewModel, mode, addedBusinessName, taxYear, changeTo, isAgent, backUrl, postUrl))
+                }
+            }
+          }
+          else{
+            Logger("application").error(
+              s"[BusinessAddedObligationsController][handleRequest] - invalid change to mode provided")
+            if (isAgent) Future(itvcErrorHandlerAgent.showInternalServerError())
+            else Future(itvcErrorHandler.showInternalServerError())
+          }
+        }
+        else{
+          Logger("application").error(
+            s"[BusinessAddedObligationsController][handleRequest] - invalid tax year provided")
+          if (isAgent) Future(itvcErrorHandlerAgent.showInternalServerError())
+          else Future(itvcErrorHandler.showInternalServerError())
+        }
+      }
+    }
+  }
+
+  def getIncomeSourceId(mode: String, id: String)(implicit user: MtdItUser[_]): Either[Throwable, String] = {
+    mode match {
```

Comment:
pls avoid use of String type to model enumeration, this is not best practice;
under Scala 2.x.x we use you should use sealed traits for this, this new type can be used within a view if you need this; + there are other benefit provided by compiler to verify if there are missed cases in your logic, this is not achievable when you use String type ..